File: accommodation-booking/server/src/EventBus.NET/SubscriptionManager/ISubscriptionManager.cs
namespace EventBus.NET.Integration.SubscriptionManager
{
    public interface ISubscriptionManager
    {
        void AddSubscription<T, TH>()
            where T : IntegrationEvent
            where TH : IIntegrationEventHandler<T>;

        bool HasSubscriptionsForEvent(string eventName);
        Type GetHandlerForEvent(string eventName);
        Type GetEventTypeByName(string eventName);
    }
}

File: accommodation-booking/server/src/Services/Accomodation/src/main/java/accomodation/integration/subscriptionManager/ISubscriptionManager.java
package accomodation.integration.subscriptionManager;

import accomodation.integration.IIntegrationEventHandler;
import accomodation.integration.IntegrationEvent;

import java.lang.reflect.Type;

public interface ISubscriptionManager {
    <T extends IntegrationEvent, TH extends IIntegrationEventHandler>
        void addSubscription(Class<T> eventType, Class<TH> eventHandlerType);
    boolean hasSubscriptionsForEvent(String eventName);
    Class getHandlerForEvent(String eventName);
    Class getEventTypeByName(String eventName);
}
